test/decimal128_test.dart

```diff
@@ -0,0 +1,305 @@
+////////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2023 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////////
+
+import 'dart:math';
+
+import 'package:meta/meta.dart';
+import '../lib/src/native/realm_core.dart';
+import 'package:test/test.dart';
+
+const int defaultTimes = 100000;
+
+void repeat(dynamic Function() body, [int times = defaultTimes]) {
+  for (var i = 0; i < times; ++i) {
+    body();
+  }
+}
+
+@isTest
+void repeatTest(String description, dynamic Function(Decimal128 x, int xInt, Decimal128 y, int yInt) body, [int times = defaultTimes]) {
+  final r = Random(42); // use a fixed seed to make tests deterministic
+  test('$description ($times variations)', () {
+    repeat(
+      () {
+        // 2^31 ensures x * y doesn't overflow
+        var xInt = r.nextInt(1 << 31);
+        final x = Decimal128.fromInt(xInt);
+        var yInt = r.nextInt(1 << 31);
+        final y = Decimal128.fromInt(yInt);
+
+        body(x, xInt, y, yInt);
+      },
+      times,
+    );
+  });
+}
+
+Future<void> main([List<String>? args]) async {
```

Comment:
we should import test.dart and `await setupTests` in all tests files. Thus at least keep consistency on init between the test files

Reply:
We don't use realms at all in `decimal128_test.dart` so I really have no reason for that. The integration of `Decimal128` with realm is tested elsewhere, like in `realm_object_test.dart`, `realm_value_test.dart`, etc.
But sure..

Reply:
I see your point, but I am afraid we might miss some initialization we rely on if we diverge here. For example the individual test timeout gets set up,  `test` method gets replace etc. For example I was surprised with my tests timing out while debugging these tests.